src/generator/transformDMMF.ts

```diff
@@ -1,19 +1,19 @@
 import type { DMMF } from '@prisma/generator-helper'
 import type { JSONSchema7, JSONSchema7Definition } from 'json-schema'
-import { TransformOptions } from './types'
-import { DEFINITIONS_ROOT } from './constants'
+import { DEFINITIONS_ROOT, JSON_SCHEMA_ID } from './constants'
 import { toCamelCase } from './helpers'
-
 import { getInitialJSON } from './jsonSchema'
 import { getJSONSchemaModel } from './model'
+import { TransformOptions } from './types'
+
 
 function getPropertyDefinition(
     model: DMMF.Model,
 ): [name: string, reference: JSONSchema7Definition] {
     return [
         toCamelCase(model.name),
         {
-            $ref: `${DEFINITIONS_ROOT}${model.name}`,
+            $ref: `${JSON_SCHEMA_ID}${DEFINITIONS_ROOT}${model.name}`,
```

Comment:
Same here

Reply:
Same reason as above.
If in original json file i have a reference and i merge it with an extended json, the new schema will think a reference in original file is referencing the extended file (which is not). Precising the $id on reference completely solves the problem.